zerver/views/upload.py

```diff
@@ -47,10 +47,24 @@ def serve_s3(request, user_profile, realm_id_str, filename, redir):
     else:
         return HttpResponseForbidden()
 
+# TODO: Rewrite this once we have django-sendfile
+def serve_local(request, path_id):
+    # type: (HttpRequest, str) -> HttpResponse
+    import os
+    import mimetypes
+    local_path = get_local_file_path(path_id)
+    if local_path is None:
+        return json_error(_("That file does not exist."), status=404)
```

Comment:
I'm not sure `json_error` is the right way to return an error that the file doesn't exist, since attempts to view a file will usually come from a browser, not an API client; maybe this should instead be written to return e.g. a standard `HttpResponseNotFound`?